tests/test_benchmarks_web_middleware.py

```diff
@@ -0,0 +1,43 @@
+"""codspeed benchmarks for web middlewares."""
+
+import asyncio
+
+from pytest_codspeed import BenchmarkFixture
+
+from aiohttp import web
+from aiohttp.pytest_plugin import AiohttpClient
+from aiohttp.typedefs import Handler
+
+
+def test_ten_web_middlewares(
+    benchmark: BenchmarkFixture,
+    loop: asyncio.AbstractEventLoop,
+    aiohttp_client: AiohttpClient,
+) -> None:
+    """Benchmark 100 requests with 10 middlewares."""
+    message_count = 100
+
+    async def handler(request: web.Request) -> web.Response:
+        return web.Response()
+
+    app = web.Application()
+    app.router.add_route("GET", "/", handler)
+
+    class MiddlewareClass:
```

Comment:
The class seems a little odd here, I don't see it's purpose.

But, I wonder if it's also worth having a benchmark with something like 10 middlewares?

Reply:
The goal is to track the overhead of processing the middleware. I added the same "empty" middleware 10 times since I didn't want to benchmark the overhead of middleware itself, only the overhead of having/executing it.

Reply:
Oh yes, I didn't look properly. Missed the append was in a loop.